tests/integrations/realcluster/etcd_key_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 TiKV Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package realcluster
+
+import (
+	"fmt"
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/tikv/pd/client/utils/testutil"
+)
+
+type etcdKeySuite struct {
+	clusterSuite
+}
+
+func TestEtcdKey(t *testing.T) {
+	suite.Run(t, &etcdKeySuite{
+		clusterSuite: clusterSuite{
+			suiteName: "etcd_key",
+		},
+	})
+}
+
+func TestMSEtcdKey(t *testing.T) {
+	suite.Run(t, &etcdKeySuite{
+		clusterSuite: clusterSuite{
+			suiteName: "etcd_key",
+			ms:        true,
+		},
+	})
+}
+
+var (
+	// The keys that prefix is `/pd`.
+	pdKeys = []string{
+		"",
+		"/pd//alloc_id",
+		"/pd//config",
+		// If not call `UpdateGCSafePoint`, this key will not exist.
+		// "/pd//gc/safe_point",
+		"/pd//gc/safe_point/service/gc_worker",
+		"/pd//keyspaces/id/DEFAULT",
+		"/pd//keyspaces/meta/",
+		"/pd//leader",
+		"/pd//member//binary_version",
+		"/pd//member//deploy_path",
+		"/pd//member//git_hash",
+		"/pd//raft",
+		"/pd//raft/min_resolved_ts",
+		"/pd//raft/r/",
+		"/pd//raft/s/",
+		"/pd//raft/status/raft_bootstrap_time",
+		"/pd//region_label/keyspaces/",
+		"/pd//rule_group/tiflash",
+		"/pd//rules/-c",
```

Comment:
What is it?

Reply:
I think it is the default rule key

https://github.com/tikv/pd/blob/be4a366da715ff3eeb628d98248068364be8ef91/pkg/schedule/placement/rule_manager.go#L43-L46

https://github.com/tikv/pd/blob/be4a366da715ff3eeb628d98248068364be8ef91/pkg/schedule/placement/rule_manager.go#L134

 that has been encoded by

https://github.com/tikv/pd/blob/be4a366da715ff3eeb628d98248068364be8ef91/pkg/schedule/placement/rule.go#L104-L107


And after removing the number from the key, it is `/pd//rules/-c`

Reply:
```golang
func TestEncodeToString(t *testing.T) {
	require.Equal(t, hex.EncodeToString([]byte("pd")), "7064")
	require.Equal(t, hex.EncodeToString([]byte("default")), "64656661756c74")
}
```

Reply:
ping @rleungx